pkg/filesystem/subvolume.go

```diff
@@ -201,17 +208,100 @@ func unMarshaljson(list string) []fsStruct {
 	if errg != nil {
 		logging.Fatal(errg)
 	}
-
 	return unmarshal
 }
 
 func Delete(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, fs, subvol, svg string) {
 	k8sSubvolume := getK8sRefSubvolume(ctx, clientsets)
 	_, check := k8sSubvolume[subvol]
 	if !check {
-		exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
+		deleteOmap(ctx, clientsets, OperatorNamespace, CephClusterNamespace, subvol, fs)
+		_, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
+		if err != nil {
+			logging.Error(err, "failed to delete subvolume of %q/%q", fs, subvol)
+		}
```

Comment:
The logging.Info needs to be in the else block
```suggestion
		} else {
```

Reply:
ahh, missed this.